### PR TITLE
use vue instead of riot

### PR DESCRIPTION
**Pull request by taka-sho**

### CODE REVIEW

File: gulpfile.babel.js [hunk 1]
```diff
@@ -145,6 +136,6 @@ gulp.task('compilePugToHtml', () => {
 });
 
 gulp.task('distModules', () => {
```
Comment (9renpoto) [inline gulpfile.babel.js]: please remove this task

File: package.json [hunk 2]
```diff
@@ -7,12 +7,14 @@
     "url": "https://github.com/uncovertruth/userdive-tag-paster/pulls"
   },
   "dependencies": {
-    "riot": "^3.3.2"
+    "vue": "^2.2.6",
+    "webpack": "^2.3.3"
```
Comment (9renpoto) [inline package.json]: devDepencdencies

File: package.json [hunk 3]
```diff
@@ -29,20 +32,24 @@
...
     "gulp-load-plugins": "^1.5.0",
-    "gulp-pug": "^3.2.0",
-    "gulp-riot": "^1.1.0",
+    "gulp-pug": "^3.3.0",
     "gulp-size": "^2.1.0",
     "gulp-sourcemaps": "^2.4.1",
     "gulp-uglify": "^2.0.0",
     "gulp-useref": "^3.1.2",
     "gulp-zip": "^4.0.0",
+    "jade": "^1.11.0",
```
Comment (9renpoto) [inline package.json]: unused

File: app/scripts.babel/popup.js [hunk 4]
```diff
@@ -1,23 +1,47 @@
...
+          el: '#info',
+          render: h => h(Info)
+        });
+      }).catch(() => {
+        throw new Error('couldn\'t set page\'s informations');
+      });
+    }
+
+    setAttr (pageInfo): Promise<void> {
+      return new Promise((resolve, reject) => {
```
Comment (9renpoto) [inline app/scripts.babel/popup.js]: reject 握りつぶすとろくなことにならないのでちゃんと処理しましょう

File: .flowconfig [hunk 5]
```diff
@@ -2,9 +2,12 @@
...
 .*/node_modules/stylelint
+.*/node_modules/vue/src
 
 [include]
 
 [libs]
 
 [options]
+module.file_ext=.vue
+module.file_ext=.js
```
Comment (9renpoto) [inline .flowconfig]: unused

File: .travis.yml [hunk 6]
```diff
@@ -1,6 +1,11 @@
 sudo: false
 language: node_js
 node_js: stable
+before_install:
+- export CHROME_BIN=chromium-browser
+- export DISPLAY=:99.0
+- sh -e /etc/init.d/xvfb start
```
Comment (9renpoto) [inline .travis.yml]: unused, because karma-chrome-launcher start chrome by myself

Comment (taka-sho) [inline .travis.yml]: i removed line 7, chrome didn't start. 
https://travis-ci.org/uncovertruth/userdive-tag-paster/builds/222748972#L274

File: test/index.test.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/* @flow */
+import assert from 'power-assert';
+import Render from '../app/scripts.babel/render';
+
+declare var describe: any;
```
Comment (9renpoto) [inline test/index.test.js]: `import { describe, it, before } from 'mocha'`

File: webpack.config.babel.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import path from 'path';
+import webpack from 'webpack';
+
+module.exports = {
+  entry: {
+    popoup: './app/scripts.babel/popup.js'
+  },
+  output: {
+    path: path.resolve(__dirname, 'app/scripts'),
+    filename: 'bundle.js'
```
Comment (9renpoto) [inline webpack.config.babel.js]: `s/bundle.js/[name].bundle.js`
https://webpack.js.org/guides/code-splitting-libraries/#multiple-entries

File: webpack.config.babel.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        options: {
+          presets: ['es2015']
```
Comment (9renpoto) [inline webpack.config.babel.js]: remove it, use `babelrc`

File: package.json [hunk 10]
```diff
@@ -53,17 +64,21 @@
     "url": "git+ssh://git@github.com/uncovertruth/userdive-tag-paster.git"
   },
   "scripts": {
-    "build": "gulp build",
+    "build": "npm-run-all build:*",
+    "build:clean": "npm run clean",
```
Comment (9renpoto) [inline package.json]: clean when `prebuild`

File: package.json [hunk 11]
```diff
@@ -53,17 +64,21 @@
     "url": "git+ssh://git@github.com/uncovertruth/userdive-tag-paster.git"
   },
   "scripts": {
-    "build": "gulp build",
+    "build": "npm-run-all build:*",
```
Comment (9renpoto) [inline package.json]: `-p`

File: app/popup.pug [hunk 12]
```diff
@@ -5,12 +5,9 @@ html
...
-    // build:js scripts/popup.js
-    script(src="scripts/popup.js")
-    // endbuild
   body
-    info
-    // build:js scripts/info.js
-    script(src="scripts/info.js")
+    #info
+      info
+    // build:js scripts/popup.bundle.js
```
Comment (9renpoto) [inline app/popup.pug]: remove this comment, because `uglify` javascript with webpack

File: app/popup.pug [hunk 13]
```diff
@@ -5,12 +5,9 @@ html
...
+    script(src="scripts/popup.bundle.js")
     // endbuild
```
Comment (9renpoto) [inline app/popup.pug]: remove it

File: app/scripts.babel/render.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
+/* @flow */
+/* eslint no-new:0 */
+'use strict';
+import Vue from 'vue';
+import Info from '../components/info';
+
+module.exports = (callback) => {
```
Comment (9renpoto) [inline app/scripts.babel/render.js]: `export default`

File: test/index.test.js [hunk 15]
```diff
@@ -0,0 +1,16 @@
+/* @flow */
+import assert from 'power-assert';
```
Comment (9renpoto) [inline test/index.test.js]: `from 'assert'`

File: test/karma.conf.js [hunk 16]
```diff
@@ -0,0 +1,55 @@
+const moduleConfig = require('../webpack.base.js').moduleConfig;
```
Comment (9renpoto) [inline test/karma.conf.js]: `const { moduleConfig, resolveConfig } = require('../webpack.base')` is simple

File: test/karma.conf.js [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+        fs: 'empty'
+      }
+    },
+    port: 3000,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    browsers: ['Chrome'],
+    singleRun: true,
+    concurrency: Infinity,
+    plugins: [
```
Comment (9renpoto) [inline test/karma.conf.js]: remove `plugins` settings, becase auto import `karma-*` plugins from package.json

File: test/karma.conf.babel.js [hunk 18]
```diff
@@ -0,0 +1,48 @@
+import { moduleConfig, resolveConfig } from '../webpack.base.js';
```
Comment (9renpoto) [inline test/karma.conf.babel.js]: `.js` extensions load default

File: test/index.test.js [hunk 19]
```diff
@@ -0,0 +1,16 @@
+/* @flow */
+import assert from 'assert';
+import Render from '../app/render';
+import { describe, it, before } from 'mocha';
+
+describe('Vue unit test', () => {
+  before(() => {
+    document.body.innerHTML = window.__html__['fixtures/test.html'];
```
Comment (9renpoto) [inline test/index.test.js]: use `fixture.load`, and `fixture.cleanup`
https://github.com/billtrik/karma-fixture#usage

File: test/karma.conf.babel.js [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+    concurrency: Infinity,
+    customLaunchers: {
+      Chrome_travis_ci: {
+        base: 'Chrome',
+        flags: ['--no-sandbox']
+      }
+    }
+  });
+  if (process.env.TRAVIS) {
+    config.browsers = ['Chrome_travis_ci'];
```
Comment (9renpoto) [inline test/karma.conf.babel.js]: is this work?
`config` is not Object, this is `Funciton`

File: test/index.test.js (hunk 19 above)
Comment (taka-sho) [inline test/index.test.js]: Sry, this project doesn't use `karma-fixture` [package.json](https://github.com/uncovertruth/userdive-tag-paster/pull/86/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R40)

File: test/karma.conf.babel.js (hunk 20 above)
Comment (taka-sho) [inline test/karma.conf.babel.js]: guess it does work.
[here](https://swizec.com/blog/how-to-run-javascript-tests-in-chrome-on-travis/swizec/6647). Look "The solution"

**Pull request merged**